src/CoreVideo/CVBuffer.cs

```diff
@@ -115,32 +115,67 @@ public void RemoveAttachment (NSString key)
+		[Deprecated (PlatformName.WatchOS, 8, 0)]
 		[DllImport (Constants.CoreVideoLibrary)]
 		extern static /* CFTypeRef */ IntPtr CVBufferGetAttachment (/* CVBufferRef */ IntPtr buffer, /* CFStringRef */ IntPtr key, out CVAttachmentMode attachmentMode);
 
+		// The new method is the same as the old one but changing the ownership from Get to Copy, so we will use the new version if possible since the
+		// older method has been deprecatd.
+		[Watch (8,0), TV (15,0), Mac (12,0), iOS (15,0)]
+		[DllImport (Constants.CoreVideoLibrary)]
+		extern static /* CFTypeRef */ IntPtr CVBufferCopyAttachment (/* CVBufferRef */ IntPtr buffer, /* CFStringRef */ IntPtr key, out CVAttachmentMode attachmentMode);
+
 // FIXME: we need to bring the new API to xamcore
 #if !MONOMAC
 		// any CF object can be attached
 		public T GetAttachment<T> (NSString key, out CVAttachmentMode attachmentMode) where T : class, INativeObject
 		{
 			if (key == null)
 				throw new ArgumentNullException ("key");
+#if IOS || __MACCATALYST__ || TVOS
+			if (UIKit.UIDevice.CurrentDevice.CheckSystemVersion (15, 0))
+#elif WATCH
+			if (WatchKit.WKInterfaceDevice.CurrentDevice.CheckSystemVersion (8, 0))
+#endif
+				return Runtime.GetINativeObject<T> (CVBufferCopyAttachment (handle, key.Handle, out attachmentMode), true);
 			return Runtime.GetINativeObject<T> (CVBufferGetAttachment (handle, key.Handle, out attachmentMode), false);
 		}
 #else
 		public NSObject GetAttachment (NSString key, out CVAttachmentMode attachmentMode)
 		{
 			if (key == null)
 				throw new ArgumentNullException ("key");
-			return Runtime.GetNSObject (CVBufferGetAttachment (handle, key.Handle, out attachmentMode));
+			if (PlatformHelper.CheckSystemVersion (12, 0))
+				return Runtime.GetNSObject (CVBufferCopyAttachment (handle, key.Handle, out attachmentMode));
+			else
+				return Runtime.GetNSObject (CVBufferGetAttachment (handle, key.Handle, out attachmentMode));
 		}
 #endif
 
+		[Deprecated (PlatformName.MacOSX, 12, 0)]
+		[Deprecated (PlatformName.iOS, 15, 0)]
+		[Deprecated (PlatformName.TvOS, 15, 0)]
+		[Deprecated (PlatformName.WatchOS, 8, 0)]
 		[DllImport (Constants.CoreVideoLibrary)]
 		extern static /* CFDictionaryRef */ IntPtr CVBufferGetAttachments (/* CVBufferRef */ IntPtr buffer, CVAttachmentMode attachmentMode);
 
+		[Watch (8,0), TV (15,0), Mac (12,0), iOS (15,0)]
+		[DllImport (Constants.CoreVideoLibrary)]
+		extern static /* CFDictionaryRef */ IntPtr CVBufferCopyAttachments (/* CVBufferRef */ IntPtr buffer, CVAttachmentMode attachmentMode);
+
 		public NSDictionary GetAttachments (CVAttachmentMode attachmentMode)
 		{
+#if IOS || __MACCATALYST__ || TVOS
+			if (UIKit.UIDevice.CurrentDevice.CheckSystemVersion (15, 0))
+#elif WATCH
+			if (WatchKit.WKInterfaceDevice.CurrentDevice.CheckSystemVersion (8, 0))
+#elif MONOMAC
+			if (PlatformHelper.CheckSystemVersion (12, 0))
+#endif
+				return (NSDictionary) Runtime.GetNSObject (CVBufferCopyAttachments (handle, attachmentMode));
```

Comment:
same, you're not specifying the ownership (using the default)

Reply:
same :)

Reply:
same :)